pkg/schema/schema.go

```diff
@@ -145,6 +162,20 @@ func getValue(node yamlmeta.Node, t Type) (interface{}, error) {
 	return t.GetDefaultValue(), nil
 }
 
+func getValidations(node yamlmeta.Node) ([]assertions.Rule, error) {
+	ann, err := processOptionalAnnotation(node, AnnotationValidation, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	validationAnn, ok := ann.(*ValidationAnnotation)
+	if !ok {
+		return nil, nil
+	}
```

Comment:
If I have to do a type check to determine if the annotation is present, it's a kind of complexity that I want to make sure I'm getting something for...

And when I review `processOptionalAnnotation()`, I see _some_ common code, but perhaps not enough to justify needing to make moves this ☝🏻 .

Suggestion: extract out of `processOptionalAnnotation()` the logic for handling `AnnotationValidation` into its own function that returns an `AnnotationValidation`... which would make this chunk unnecessary.